### PR TITLE
refactor deepfloyd if img2img pipeline tests to the new mixin structure

### PR DESCRIPTION
**Pull request by akshan-main**
# What does this PR do?

Refactors the DeepFloyd IF img2img pipeline tests to the new mixin structure from #14113, same shape as the text-to-image port: an `IFImg2ImgPipelineTesterConfig` plus `TestIFImg2ImgPipeline(PipelineTesterMixin)` and `TestIFImg2ImgPipelineMemory(PipelineOffloadTesterMixin)`, slow class ported to pytest, dummy components inlined from the shared package mixin, and a fresh CPU slice in `test_inference`.

The same verified choices as #14580 apply: `@skip_mps` kept (all-NaN output on mps), `latents` dropped from `optional_input_params`, the custom `test_save_load_optional_components` ported (prompt pre-encoded), the obsolete `save_load_local` override dropped, and the float16 1e-1 tolerance kept. The old `test_float16_inference` override maps to the framework's `test_half_precision_inference_no_nan`, which needs no tolerance. Follows #14580.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Discussed on Slack with @sayakpaul
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

## Who can review?

@sayakpaul

### CODE REVIEW

File: tests/pipelines/deepfloyd_if/test_if_img2img.py [hunk 1]
```diff
@@ -36,88 +38,174 @@
...
+        generated_slice = torch.cat([generated_slice[:8], generated_slice[-8:]])
+        assert_tensors_close(generated_slice, expected_slice, atol=1e-3)
 
-    def test_save_load_local(self):
-        self._test_save_load_local()
+    @pytest.mark.skipif(torch_device not in ["cuda", "xpu"], reason="float16 requires CUDA or XPU")
+    @require_accelerator
+    def test_save_load_float16(self, tmp_path):
+        # Due to non-determinism in save load of the hf-internal-testing/tiny-random-t5 text encoder
+        super().test_save_load_float16(tmp_path, expected_max_diff=1e-1)
```
Comment (sayakpaul) [inline tests/pipelines/deepfloyd_if/test_if_img2img.py]: I think if we load it using `eval()` it should be fine? Could we check?

**Review by sayakpaul — COMMENTED** [review]
Thanks. Left one comment.

File: tests/pipelines/deepfloyd_if/test_if_img2img.py (hunk 1 above)
Comment (akshan-main) [reply tests/pipelines/deepfloyd_if/test_if_img2img.py]: You're right. The old base test ran the fresh components in train mode, so the tiny-random-t5 dropout was the "non-determinism"; the new get_pipeline() puts both sides in eval. Replicated the save/load flow on CPU fp32: eval both sides gives max_diff 1.7e-5, the train-mode first pipe gives 0.94. Dropped the override here and in #14580.

**Review by sayakpaul — APPROVED** [review]

**Review by sayakpaul — APPROVED** [review]
